Поиск с параметрами

lalka2016 opened the pull request:
Данный пул реквест добавляет сортировку поиска (#57) и поиск по параметрам (#662, #859), а так же немного меняет его дизайн. 
Ещё теперь можно искать видео, посты (#384), комментарии и приложения.
![изображение](https://github.com/openvk/openvk/assets/99399973/be93ab91-eadb-4dd0-a6ef-2e038d95e8d6)
Взял с:
![изображение](https://github.com/openvk/openvk/assets/99399973/bc456946-e915-41ab-8331-df129b5ac2aa)


Сортировка поиска в две стороны, сортировать можно по id и по рандому, а если ищите людей, то по рейтингу и имени. Фильтровать можно по дате создания чего-либо (кроме групп и приложений, у них не сохраняется дата создания), если же ищите людей — то по статусу, городу, полу, семейному положению, полит.взглядам, контактам и интересам, а так же по наличию в онлайне (#768). 
![изображение](https://github.com/openvk/openvk/assets/99399973/0b325f08-8973-4fdb-8e5c-1ef09be91fec)


Ещё интересы стали кликабельными (их надо разделять запятой и пробелом)

![изображение](https://github.com/openvk/openvk/assets/99399973/182ce840-db0f-4e88-8656-a2a38ddbc1b0)


Однако у этого пулл реквеста есть большая проблема — отображение постов из групп где включена настройка "не показывать в глобальной ленте". Я исправил это костыльным образом: если это пост из подобной группы, то он просто не показывается. Однако допустим, что подобная группа сделала 10 постов, и у нас получится пустая страница. 
![изображение](https://github.com/openvk/openvk/assets/99399973/77a2e81f-c101-4e20-8d86-c94ab7671c9c)
Если вы знаете, как это можно нормально сделать — сделайте пулл в ветку

Commit:
Search with parameters

Commit:
Small improvements

Commit:
Small improvements no.2

Поиск теперь нормально выглядит на других темах
При поиске по группам отображается количество участников там
Костыль с пропуском постов возвращён но немного изменён
Добавлен костыль к комментариям чтобы не показывались комменты с удалённых постов и не было бага как в вепуровке
Добавлены ключи в советском и имперском языке для моих прошлых пуллов (ну и этого конечно)

Commit:
Fix debilny oshibky

Убрал лишние ключи локализации, исправил панель поеска и исправил hometown

veselcraft reviewed (changes requested):
ну вроде норм но есть пару momentov)

Commit:
fiksy bagiv

veselcraft reviewed (approved):
претензий нема кроме растровых пикч и ксс, но это я с @myslivets сами исправим, можешь не париться

Commit:
Merge branch 'master' into coolSearchNoGovnocode

celestora reviewed (commented):
кодстайл местами отличается от кодстайла опенвк а так в принцпе хз давайте

maksalees reviewed (changes requested):
Няшно

Commit:
ok

Commit:
Merge branch 'coolSearchNoGovnocode' of https://github.com/lalka2016/openvkuh into coolSearchNoGovnocode

Commit:
ok 2

Commit:
ok

Commit:
rrrrrrrrrrrrrrrr

Добавил параметры в vkapi users.search(), пофиксил проверку коммерции, сузил параметры ещё больше и добавил анимацию выдвижения поиска чтобы красиво было

celestora reviewed (approved):
https://music.youtube.com/watch?v=WmC66JLxYYw
